internal/gatewayapi/backendtrafficpolicy.go

```diff
@@ -0,0 +1,338 @@
+		routeMap[types.NamespacedName{
+			Name:      route.GetName(),
+			Namespace: route.GetNamespace(),
+		}] = route
+	}
+	gatewayMap := map[types.NamespacedName]*GatewayContext{}
+	for _, gw := range gateways {
+		gatewayMap[types.NamespacedName{
+			Name:      gw.GetName(),
+			Namespace: gw.GetNamespace(),
+		}] = gw
+	}
+
+	targetedGateways := make(map[types.NamespacedName]sets.Set[string])
+	targetedRoutes := map[routePolicyTarget]bool{}
+
+	// Keep track of all the parent gateways of routes that are being targeted by these policies so that we can set the
+	// overridden status on them if needed
+	overriddenPolicyTargets := map[types.NamespacedName]bool{}
+
+	// Translate
+	// 1. First translate Policies targeting xRoutes
+	// 2. Next translate Policies targeting Gateways with a sectionName
+	// 3. Finally, the policies targeting Gateways but without a sectionName
+
+	// TODO: Import sort order to ensure policy with same section always appear
+	// before policy with no section so below loops can be flattened.
+	// This function is messy due to all the context passing/parsing to set the appropriate conditions for all the policies.
+	// There is definitely room for some cleanup and optimization here.
+
+	// Process the policies targeting xRoutes
+	for _, policy := range backendTrafficPolicies {
+		if policy.Spec.TargetRef.Kind != KindGateway {
+			policy := policy.DeepCopy()
+			res = append(res, policy)
+
+			// Negative statuses have already been assigned so its safe to skip
+			target := resolveBTPolicyTargetRef(policy, gatewayMap, routeMap)
+			if target == nil {
+				continue
+			}
+			routeKey := routePolicyTarget{key: *target, kind: policy.Spec.TargetRef.Kind}
+
+			// Check if another policy targeting the same xRoute exists
+			if _, ok := targetedRoutes[routeKey]; ok {
+				message := fmt.Sprintf("Unable to target %s, another BackendTrafficPolicy has already attached to it", routeKey.kind)
+
+				status.SetBackendTrafficPolicyCondition(policy,
+					gwv1a2.PolicyConditionAccepted,
+					metav1.ConditionFalse,
+					gwv1a2.PolicyReasonConflicted,
+					message,
+				)
+
+				continue
+			}
+
+			targetedRoutes[routeKey] = true
+			route := routeMap[*target]
+			for _, parentRef := range GetParentReferences(route) {
+				namespace := ""
+				if parentRef.Namespace != nil {
+					namespace = string(*parentRef.Namespace)
+				}
+				overriddenPolicyTargets[types.NamespacedName{
+					Name:      string(parentRef.Name),
+					Namespace: namespace,
+				}] = true
+			}
+			translateBackendTrafficPolicy(policy, xdsIR)
+
+			// Set Accepted=True
+			status.SetBackendTrafficPolicyCondition(policy,
+				gwv1a2.PolicyConditionAccepted,
+				metav1.ConditionTrue,
+				gwv1a2.PolicyReasonAccepted,
+				"BackendTrafficPolicy has been accepted.",
+			)
+		}
+	}
+
+	// Process the policies targeting Gateways with a section name
+	for _, policy := range backendTrafficPolicies {
+		if policy.Spec.TargetRef.SectionName != nil {
```

Comment:
in theory a route will never reach here because of the CEL validation in the CRD, but this maybe hard for the next dev looking at this to figure that out
can we add an extra checking in here
```
policy.Spec.TargetRef.Kind == KindGateway &&  .....
```